node/forest_libp2p/src/rpc/mod.rs

```diff
@@ -142,12 +118,45 @@ where
     where
         T: AsyncWrite + Unpin + Send,
     {
-        // TODO: Use FramedWrite to stream write. Dilemma right now is if we should fork the cbor codec so we can replace serde_cbor to our fork of serde_cbor
-        io.write_all(
-            &to_vec(&res).map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?,
-        )
-        .await?;
-        io.close().await?;
-        Ok(())
+        encode_and_write(io, res).await
     }
 }
+
+async fn read_and_decode<IO, T>(io: &mut IO) -> io::Result<T>
+where
+    IO: AsyncRead + Unpin,
+    T: serde::de::DeserializeOwned,
+{
+    const MAX_BYTES_ALLOWED: usize = 2 * 1024 * 1024; // messages over 2MB are likely malicious
+    const TIMEOUT_SECS: u64 = 30;
+
+    // Currently the protocol does not send length encoded message,
```

Comment:
I believe currently if some malicious nodes send cbor encoded bytes + some random bytes in a single request/response, it will deadlock forest daemon because `FramedRead` internally ensures no trailing data, failure to detect end of frame will continue the loop and end up hitting deadlock at the next `io.poll_ready`